Bug with mix + elem

From v6.5.2 there is bug:

```js
  it.only('should work!', function() {
    test(function() {
      block('b').elem('e').replace()(function() {
        var mix = [ {
            block: this.block,
            elem: this.elem,
            mods: this.mods
          } ].concat(this.ctx.mix);

        return {
          block: 'replace',
          mix: mix
        };
      });
    }, {
      block: 'b',
      elem: 'e',
      mix: { block: 'bemjson' }
    }, '<div class="replace b__e bemjson"></div>');
  });
```

```js
      AssertionError: '<div class="replace b__e bemjson__e bemjson__e"></div>' deepEqual '<div class="replace b__e bemjson"></div>'
      + expected - actual

      -<div class="replace b__e bemjson__e bemjson__e"></div>
      +<div class="replace b__e bemjson"></div>
```

Also in v4.3.5.

--- a/.jshintrc
+++ b/.jshintrc
@@ -29,7 +29,7 @@
     "unused"        : true,     // true: Require all defined variables be used
     "strict"        : false,    // true: Requires all functions run in ES5 Strict Mode
     "maxparams"     : false,    // {int} Max number of formal params allowed per function
-    "maxdepth"      : 3,        // {int} Max depth of nested blocks (within functions)
+    "maxdepth"      : 4,        // {int} Max depth of nested blocks (within functions)
     "maxstatements" : false,    // {int} Max number statements per function
     "maxcomplexity" : false,    // {int} Max cyclomatic complexity per function
     "maxlen"        : false,    // {int} Max number of characters per line

--- a/lib/bemhtml/runtime/index.js
+++ b/lib/bemhtml/runtime/index.js
@@ -606,6 +606,8 @@ BEMHTML.prototype.renderMix = function renderMix(entity,
       if (!nestedItem.block &&
           !nestedItem.elem ||
           !visited[classBuilder.build(nestedItem.block, nestedItem.elem)]) {
+        if (nestedItem.block) continue;
+
         nestedItem._block = block;
         nestedItem._elem = elem;
         mix = mix.slice(0, i + 1).concat(

--- a/test/runtime-test.js
+++ b/test/runtime-test.js
@@ -401,6 +401,26 @@ describe('BEMHTML compiler/Runtime', function() {
          '</a>');
     });
 
+    it('should render both mix from templates and from bemjson', function() {
+      test(function() {
+        block('b').elem('e').replace()(function() {
+          var mix = [
+            { block: this.block, elem: this.elem },
+            { block: 'sprite', mods: { test: 'opa' } }
+          ].concat(this.ctx.mix);
+
+          return {
+            block: 'replace',
+            mix: mix
+          };
+        });
+      }, {
+        block: 'b',
+        elem: 'e',
+        mix: { block: 'bemjson' }
+      }, '<div class="replace b__e sprite sprite_test_opa bemjson"></div>');
+    });
+
     it('should check that mix do not overwrite jsParams', function() {
       test(function() {
         block('b1')(

